concourse/steps/scan_sources.py

```diff
@@ -63,98 +66,156 @@ def scan_component_with_whitesource(
 
     # create whitesource client
     ci.util.info('creating whitesource client')
-    client = whitesource.util.create_whitesource_client(whitesource_cfg_name=whitesource_cfg_name)
+    ws_client = whitesource.util.create_whitesource_client(
+        whitesource_cfg_name=whitesource_cfg_name,
+    )
 
     # parse component_descriptor
     ci.util.info('parsing component descriptor')
     component_descriptor = cm.ComponentDescriptor.from_dict(
         ci.util.parse_yaml_file(component_descriptor_path)
     )
 
-    # for component in []: 'components matching component_descriptor'
-    for component in product.v2.components(component_descriptor_v2=component_descriptor):
+    product_name = component_descriptor.component.name
```

Comment:
in a subsequent change, we might want to mv the naming-logic (component.name -> product_name) out of the template (into a util- or model-function in whitesource-pkg). At least in case we rely on a certain naming scheme, and this becomes part of a contract we use.